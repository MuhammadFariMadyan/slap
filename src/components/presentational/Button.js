import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

const Container = styled.div`
  display: inline-flex;
  width: auto;
  align-items: center;
  background: #6ab04c;
  color: white;
  border: 0px;
  padding: 8px 12px;
  font-size: 11px;
  outline: none;
  user-select: none;
  cursor: pointer;

  &:hover {
    background: #4a8232;
  }
`;

const Caption = styled.p`
  font-size: 12px;
  font-family: Roboto;
  margin: 0px;
`;

const Button = ({ className, icon, caption, onClick }) => (
  <Container className={className} onClick={onClick}>
    {icon && icon}
    <Caption>{caption}</Caption>
  </Container>
);

Button.propTypes = {
  className: PropTypes.string,
  caption: PropTypes.string,
  icon: PropTypes.object,
  onClick: PropTypes.func
};

export default Button;
