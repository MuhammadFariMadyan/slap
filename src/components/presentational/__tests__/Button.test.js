import React from "react";
import { shallow } from "enzyme";

import Button from "../Button";

describe("Button", () => {
  const Dummy = () => <div />;

  const setup = propOverrides => {
    const props = {
      caption: "My Own Button",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Button {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with icon", () => {
    const { wrapper } = setup({ icon: <Dummy /> });

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
