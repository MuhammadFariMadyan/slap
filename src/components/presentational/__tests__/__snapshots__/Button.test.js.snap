// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.p>
    My Own Button
  </styled.p>
</styled.div>
`;

exports[`Button should renders with icon 1`] = `
<styled.div
  onClick={[Function]}
>
  <Dummy />
  <styled.p>
    My Own Button
  </styled.p>
</styled.div>
`;
